spec/factories/miq_group.rb

```diff
@@ -5,17 +5,23 @@
     transient do
       features nil
       role nil
+
+      # Temporary: Allows entitlements, a new table, to be invisible to current spec setup code
+      # Do NOT use these if possible.
+      miq_user_role_id nil
+      miq_user_role nil
     end
 
     sequence(:sequence)  # don't want to spend time looking these up
     description { |g| g.role ? "EvmGroup-#{g.role}" : generate(:miq_group_description) }
 
     after :build do |g, e|
-      if e.role
-        g.miq_user_role = MiqUserRole.find_by_name("EvmRole-#{e.role}") ||
-                          FactoryGirl.create(:miq_user_role, :features => e.features, :role => e.role)
-      elsif e.features.present?
-        g.miq_user_role = FactoryGirl.create(:miq_user_role, :features => e.features)
+      if e.role || e.features || e.miq_user_role_id || e.miq_user_role
+        g.entitlement = FactoryGirl.create(:entitlement,
+                                           :features => e.features,
+                                           :role => e.role,
+                                           :miq_user_role_id => e.miq_user_role_id,
+                                           :miq_user_role => e.miq_user_role)
```

Comment:
Yeah I don't like using these too much but the case for making users with roles quickly is understandable. I don't like what I added here at all, but it keeps me form having to rewrite untold numbers of specs to get this tenancy epic moving and open it up to other developers.

I _do_ want to return and remove these eventually.